src/univariate/continuous/triangular.jl

```diff
@@ -119,27 +133,22 @@ function quantile(d::TriangularDist, p::Real)
               b - sqrt(b_m_a * (b - c) * (1 - p))
 end
 
-function mgf(d::TriangularDist{T}, t::Real) where T<:Real
-    if t == zero(t)
-        return one(T)
-    else
-        (a, b, c) = params(d)
-        u = (b - c) * exp(a * t) - (b - a) * exp(c * t) + (c - a) * exp(b * t)
-        v = (b - a) * (c - a) * (b - c) * t^2
-        return 2u / v
-    end
+function mgf(d::TriangularDist, t::Real)
+    a, b, c = params(d)
+    T = typeof(t / a)
+    t == zero(T) && return one(T)
+    u = (b - c) * exp(a * t) - (b - a) * exp(c * t) + (c - a) * exp(b * t)
+    v = (b - a) * (c - a) * (b - c) * t^2
+    return 2u / v
```

Comment:
The function is not type stable since sometimes `T` is not a good heuristic for the type of the result here.
```julia
julia> @inferred mgf(TriangularDist{Int}(0, 2, 1), 1//1)
ERROR: return type Float64 does not match inferred return type Union{Rational{Int64}, Float64}
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] top-level scope
   @ REPL[22]:1
```

Reply:
updated